cmd/frontend/main.go

```diff
@@ -154,7 +162,10 @@ func main() {
 		)
 
 		server := &http.Server{Addr: *listenAddress}
+		buildInfoHandler := http.HandlerFunc(promapi.BuildinfoHandlerFunc(log.With(logger, "component", "buildinfo-handler"), "frontend", version))
+		http.Handle("/api/v1/status/buildinfo", buildInfoHandler)
 		http.Handle("/metrics", promhttp.HandlerFor(metrics, promhttp.HandlerOpts{Registry: metrics}))
+		http.Handle("/api/v1/rules/test/test", http.NotFoundHandler()) // Magical endpoint used by Grafana to check whether a datasource has the ruler-api enabled. Should contain the words "page not found" in the response.
```

Comment:
Nice! Should we instead ensure that all unknown paths respond with `NotFoundHandler`? 🤔 Sounds like a required part of rule API compatibility here?